clang/lib/Sema/SemaInit.cpp

```diff
@@ -8340,8 +8340,17 @@ void Sema::checkInitializerLifetime(const InitializedEntity &Entity,
             << Entity.getType()->isReferenceType() << CLE->getInitializer() << 2
             << DiagRange;
       } else {
-        Diag(DiagLoc, diag::warn_ret_local_temp_addr_ref)
-         << Entity.getType()->isReferenceType() << DiagRange;
+        // P2748R5: Disallow Binding a Returned Glvalue to a Temporary.
+        // [stmt.return]/p6: In a function whose return type is a reference,
+        // other than an invented function for std::is_convertible ([meta.rel]),
+        // a return statement that binds the returned reference to a temporary
+        // expression ([class.temporary]) is ill-formed.
```

Comment:
Did you try to use `is_convertible` with this change? This will probably need tests too

Reply:
Oh, you did!

Reply:
Do we actually cover all the cases in https://cplusplus.github.io/CWG/issues/2826.html 
@t3nsor ?

Reply:
> Did you try to use `is_convertible` with this change? This will probably need tests too

Yeah, I've added 2 simple cases, I'd like to add more case if needed. I wanted to implement LWG4028 as well, but it looks like it's still in the draft stage, and IIUC, if `From` is a reference the following wording seems no real impact to Clang, but except https://godbolt.org/z/bW3Wsa5PW
```
The predicate condition for a template specialization is_convertible<From, To> shall be 
satisfied if and only if the return expression in the following code would be well-formed, 
including any implicit conversions to the return type of the function:
To test() {
  return declval<From>();
}
```

Reply:
> I wanted to implement LWG4028 as well

I don't know which part of LWG4028 is "not yet implemented". As the submitter, my intent was just to reflect what implementations actually did and to avoid crux in the core wording.

Reply:
Great, this example makes things much more clearer!

Reply:
Sorry, I misunderstanding before. I have seen [LWG3400](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p2748r5.html#ref-LWG3400), and there has a comments `This issue should be resolved by LWG [4028](https://cplusplus.github.io/LWG/lwg-active.html#4028).`, therefor, what I mean here is that should we makes `std::is_nothrow_convertible_v<int, Weird>` evaluated as `true`?

Reply:
I don't know if Clang implements all the cases in [class.temporary], but I would assume "yes", because if not, then users are already not getting lifetime-extended temporaries in places where they would expect them, which would be bad. The changes to this file are basically the same as what I had prototyped in my branch.

Reply:
> should we makes `std::is_nothrow_convertible_v<int, Weird>` evaluated as `true`?

I don't see any proper motivation for doing so. Implementations consistently match the design in [P0758R1](https://wg21.link/p0758r1). It's the standard wording that failed to reflect the intent, which is perhaps a defect. (Additionally, the pasted sample implementation would exactly reflect the bug in the `noexcept` operator, if any.)

And IMO `std::is_nothrow_convertible(_v)` should be implementable without intrinsic (and the implementation strategy would be [quite simple](https://en.cppreference.com/w/cpp/types/is_convertible#Possible_implementation)). The `__is_nothrow_convertible` instrinsic should only be meaningful for acceleration of compilation.

Reply:
> I don't know if Clang implements all the cases in [class.temporary]

AFAIK, I think Clang covered all the cases in [class.temporary](except https://github.com/llvm/llvm-project/pull/87933 and https://github.com/llvm/llvm-project/pull/86960 they still WIP).

> And IMO std::is_nothrow_convertible(_v) should be implementable without intrinsic (and the implementation strategy would be [quite simple](https://en.cppreference.com/w/cpp/types/is_convertible#Possible_implementation)). The __is_nothrow_convertible instrinsic should only be meaningful for acceleration of compilation.

Wow! it's sensible to me now, somehow, I've a local patch to build a hypothetical function declared `void conv-dest(To) noexcept;`.............., I don't know why I would do such a thing, lol.